scripts/scrapper.ts

```diff
@@ -1,10 +1,44 @@
 import axios from 'axios'
 import * as cheerio from 'cheerio'
-import { MealCategory } from '#models/meal'
+import Meal, { MealCategory } from '#models/meal'
+import { createHash } from 'node:crypto'
+import db from '@adonisjs/lucid/services/db'
+import WebsiteHash from '#models/website_hash'
+import logger from '@adonisjs/core/services/logger'
 
-export async function scrapeMenu() {
-  const url = 'https://sks.pwr.edu.pl/menu/'
+const url = 'https://sks.pwr.edu.pl/menu/'
+
+export async function runScrapper() {
+  const currentHash = await cacheMenu()
+  const storedHash = await WebsiteHash.first()
+
+  const trx = await db.transaction()
+  try {
+    if (!storedHash) {
```

Comment:
so I'm not sure what type it is here, but I assume `string | null | undefined` or smth similar?
so this maybe not thaat problematic in ts like in js, but generally speaking: it's better to use more specific checks than `!storedHash`. E.g. it's better to check `storedHash !== null && storedHash !== undefined `

Why? Cause js/ts casting to bools are funny and e.g. an  empty string are falsy too. So it's better to be specific.

Reply:
take note I have no idea what this thing's type

Reply:
Oh I see, I guess it's an object

Reply:
but still it's prolly better to be specific with null checks (afaik)

Reply:
yeah, the type is `WebsiteHash | null`

kinda wanted to do it this way, because of this string boolean logic (empty string should be treated as no hash)

is it still worth to be explicit in this case?

Reply:
in such case then probably a matter of preference comes into play

Reply:
gpt incoming:

```
If the type is WebsiteHash | null and you want an empty string to be treated as "no hash," then using if (!storedHash) is a practical and concise approach, as it captures both null and the empty string cases. In this situation, the shorthand may be preferable since it’s both:

Concise: It checks for both null and "" without extra conditions.
Intentional: It aligns with your logic to treat any falsy value as "no hash."
When to Still Consider Being Explicit
However, there are a few reasons you might still prefer an explicit check, like storedHash === null || storedHash === "":

Type Clarity: It can be clearer to future readers (or yourself) that both null and "" are being explicitly handled as empty hash cases. This way, someone reading the code later understands the intent better.

Refactoring Safety: If the type definition (WebsiteHash | null) later changes (for instance, to include undefined), an explicit check would prevent undefined from being treated as a "no hash" case by accident.

In this case, though, it seems like if (!storedHash) is a good choice if you're confident that:

Only null and "" should be handled as "no hash."
This approach aligns with the function’s purpose.
So yes, it does come down to preference! If you want to minimize verbosity and don’t anticipate needing more precise checks, if (!storedHash) is likely to work well for your needs.
```

Reply:
but noo

Reply:
empty hash won't be treated as no hash in this case

Reply:
cause it's an object and not a string so it'll be {hash: ""} which is truthy I guess

Reply:
So imo I'd be for more specific checking as it's just clearer what you mean by this check

Reply:
okey dokey